feat(curriculum): add rock paper scissors workshop (#55957)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/workshop-rps-game/66cf35a75f891ab4f4e5497b.md

---
id: 66cf35a75f891ab4f4e5497b
title: Step 11
challengeType: 0
dashedName: step-11
---

# --description--

> If you try to play the game, you will see that you can play for an infinite amount of rounds. But the rules state that the first one to three points wins.
> 
> Inside your `showResults` function, you will need to check if the player or computer has reached three points. If either has reached three points, you should display a message indicating the winner. 
> 
> For example, if the player has won the game, then the `winnerMsgElement` should be updated to `"Player has won the game!"`. If the computer has won the game, then the `winnerMsgElement` should be updated to `"Computer has won the game!"`.
> 
> If there is a winner, you will want to show the `resetGameBtn` button and hide the `optionsContainer` so the player can play again.

Reviewer: ```suggestion
If you try to play the game, you will see that you can play for an infinite amount of rounds. But the rules state that the first one to three points wins. You want to check if there's a winner, and display a message.

In your `showResults` function, if the player has reached three points, update the `winnerMsgElement` to `"Player has won the game!"`. If the computer has reached three points, update the `winnerMsgElement` to `"Computer has won the game!"`.

If there is a winner, show the `resetGameBtn` button by settings it's `display` to `block` and hide the `optionsContainer` by setting it's `display` to `none`.
```

I struggled with this one a bit, the instructions weren't all that clear to me - I came up with a suggestion.
